service/src/main/resources/example.application.yaml

```diff
@@ -29,6 +29,7 @@ loginsvc:
             - username: "user1"
               password: "password1"
               groups:
+                - "group1"
```

Comment:
Let me try and figure this out... I think from the Pureconfig perspective an empty tag is the same as not having the tag there at all so it would have to be an option... but I think I can find a way around it.

Reply:
Ok, so it seems for PureConfig to be able to ensure that the field is required but it's contents can be empty, I need to set it like so:
groups: Array[Option[String]]

And the Config should look like this to be empty:
            - username: "user1"
              password: "password1"
              groups:
                -

Reply:
what about 
```
- username: "user1"
password: "password1"
groups: []
```
? (from top of my head, might not work)

`groups: Array[Option[String]]` looks totally unreasonable to me

Reply:
`- username: "user1"
password: "password1"
groups: []`

This works

`groups: Array[Option[String]]` Not sure how to get around this since PureConfig seems to need values to be typed as Options for them to be considered optional.

Reply:
> - username: "user1" password: "password1" groups: [] .... This works

with `groups: Array[String]`?

Reply:
No, with the Option

Currently trying with
`groups: Array[String]`

Reply:
It works.

Changing it back

Reply:
Changed it back

So this works:

Config:
`- username: "user1"
password: "password1"
groups: []`

Typing:
`groups: Array[String]`

Reply:
⬆️ sounds good to me.

Another observation I forgot to mention - when I tried it last time, the weirdest thing was that there was no error about invalid config at all, the whole auth provider was ignored (refused to authenticate even users with groups etc)

Reply:
I think since it's because the auth methods are optional, so if the auth method is unparsable then it takes it as if it wasn't there in the 1st place. I can add logging to indicate if a auth method is being skipped?